src/main/java/bean/Bean.java

```diff
@@ -0,0 +1,143 @@
+                    case "todo":
+                        tasks.addTask(new TodoTask(command.getDetails()));
+                        ui.showTaskAdded(tasks.getTask(tasks.size() - 1), tasks.size());
+                        break;
+                    case "deadline":
+                        String[] deadlineParts = command.getDetails().split(" /by ");
+                        tasks.addTask(new DeadlineTask(deadlineParts[0], deadlineParts[1]));
+                        ui.showTaskAdded(tasks.getTask(tasks.size() - 1), tasks.size());
+                        break;
+                    case "event":
+                        String[] eventParts = command.getDetails().split(" /from | /to ");
+                        tasks.addTask(new EventTask(eventParts[0], eventParts[1], eventParts[2]));
+                        ui.showTaskAdded(tasks.getTask(tasks.size() - 1), tasks.size());
+                        break;
+                    case "find":
+                        List<Task> foundTasks = tasks.findTasks(command.getDetails());
+                        ui.showMatchingTasks(foundTasks);
+                        break;
+                    default:
+                        ui.showError("Unknown command");
+                        break;
+                }
+                storage.save(tasks.getTasks());
+            } catch (Exception e) {
+                ui.showError(e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Checks if the given index is valid for the task list.
+     *
+     * @param index The index to be checked.
+     * @return True if the index is valid, false otherwise.
+     */
+    private boolean isValidIndex(int index) {
+        if (index < 0 || index >= tasks.size()) {
+            ui.showError("bean.task.Task index is out of bounds.");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Handles operations on tasks like marking, unmarking, and deleting.
+     *
+     * @param commandType The type of operation to perform.
+     * @param index The index of the task to operate on.
+     */
+    private void handleTaskOperation(String commandType, int index) {
+        if (isValidIndex(index)) {
+            switch (commandType) {
+            case "mark":
+                tasks.markTask(index);
+                ui.showTaskMarked(tasks.getTask(index));
+                break;
+            case "unmark":
+                tasks.unmarkTask(index);
+                ui.showTaskUnmarked(tasks.getTask(index));
+                break;
+            case "delete":
+                Task removedTask = tasks.deleteTask(index);
+                ui.showTaskDeleted(removedTask, tasks.size());
+                break;
+            }
+        }
+    }
+
+    /**
+     * The main method that starts the application.
+     *
+     * @param args Command line arguments (not used).
+     */
+    public static void main(String[] args) {
+        new Bean("data/bean.txt").startConversation();
+    }
```

Comment:
```suggestion
private static final String FILE_PATH = "./data/bean.txt"
    public static void main(String[] args) {
        new Bean("FILE_PATH").startConversation();
    }
```

I would suggest that you declare a constant variable for the file path as it makes it clear what was being passed as the parameter.